public/src/components/channelManagement/testEditor.tsx

```diff
@@ -41,279 +40,240 @@ interface DataFromServer<T extends Test> {
  */
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 const TestEditor = <T extends Test>(
   InnerComponent: React.ComponentType<InnerComponentProps<T>>,
   settingsType: FrontendSettingsType,
-) =>
-  class extends React.Component<{}, TestFormState<T>> {
-    state: TestFormState<T> = {
-      tests: null,
-      version: null,
-      selectedTestName: undefined,
-      editMode: false,
-      lockStatus: { locked: false },
-      modifiedTests: {},
-      timeoutAlertId: null,
-    };
+): React.FC => {
+  const WrappedInnerComponent: React.FC = () => {
+    const [tests, setTests] = useState<T[] | null>(null);
+    const [version, setVersion] = useState<string | null>(null);
+    const [editMode, setEditMode] = useState<boolean>(false);
+    const [selectedTestName, setSelectedTestName] = useState<string>('');
+    const [selectedTestIsValid, setSelectedTestIsValid] = useState<boolean>(true);
+    const [selectedTestHasBeenModified, setSelectedTestHasBeenModified] = useState<boolean>(true);
+    const [lockStatus, setLockStatus] = useState<LockStatus>({ locked: false });
 
-    UNSAFE_componentWillMount(): void {
-      this.fetchStateFromServer();
-    }
+    useEditModeAlertTimer(editMode);
 
-    fetchStateFromServer = (): void => {
+    const fetchStateFromServer = (): void => {
       fetchFrontendSettings(settingsType).then((serverData: DataFromServer<T>) => {
         const editMode = serverData.status.email === serverData.userEmail;
 
-        this.updateWarningTimeout(editMode);
-
-        this.setState({
-          ...serverData.value,
-          version: serverData.version,
-          lockStatus: serverData.status,
-          editMode: editMode,
-          modifiedTests: {},
-        });
+        setTests(serverData.value.tests);
+        setVersion(serverData.version);
+        setEditMode(editMode);
+        setLockStatus(serverData.status);
+        setSelectedTestIsValid(true);
+        setSelectedTestHasBeenModified(false);
       });
     };
 
-    // Maintains an alert if tool is left open for edit for 20 minutes
-    updateWarningTimeout = (editMode: boolean): void => {
-      if (editMode) {
-        if (this.state.timeoutAlertId) {
-          window.clearTimeout(this.state.timeoutAlertId);
-        }
+    useEffect(() => {
+      fetchStateFromServer();
+    }, []);
 
-        const timeoutAlertId = window.setTimeout(() => {
-          alert(
-            "You've had this editing session open for 20 minutes - if you leave it much longer then you may lose any unsaved work!\nIf you've finished then please either save or cancel.",
-          );
-          this.setState({ timeoutAlertId: null });
-        }, 60 * 20 * 1000);
-
-        this.setState({ timeoutAlertId });
-      } else if (this.state.timeoutAlertId) {
-        window.clearTimeout(this.state.timeoutAlertId);
-        this.setState({ timeoutAlertId: null });
-      }
+    const save = async (updatedTests: T[]): Promise<void> => {
```

Comment:
does this need to be `async`?

Reply:
Originally I had 
```js
    const save = (updatedTests: T[]): Promise<void> => {
      const postData = {
        version: version,
        value: {
          tests: updatedTests,
        },
      };

      return saveFrontendSettings(settingsType, postData)
        .then(resp => {
          if (!resp.ok) {
            resp.text().then(msg => alert(msg));
          }
        })
        .catch(() => {
          alert('Error while saving');
        })
        .then(fetchStateFromServer);
    };
```
But then vs code gave me a squiggly underline saying it could be made async - I'm not 100% sure if that's coming from how I have vs code set up or if that's part of the linting on the project

Reply:
Weird, I just tried it in intellij and it's happy.
Returning a Promise from a non-async function is perfectly fine. I don't see the benefit in marking something as async if you're not using await.

Reply:
I think it's a vs code thing - I'll remove the async 👍